dev/app-shell/elements/arc-footer.html

```diff
@@ -57,9 +60,55 @@
     }
     _onPlanSelected(e, suggestion) {
       this._fire('suggest', suggestion);
+      this._doBackendSearch(null);
+    }
+    _onSearchClick(e) {
+      this._updateSearchProps('*');
+    }
+    _onSearchChange(e) {
+      let search = e.target.value.toLowerCase();
+      if (!search || search == '*' || search[search.length - 1] == ' ') {
+        this._doBackendSearch(search);
+      } else {
+        this._updateSearchProps(search);
+      }
+    }
+    _onSeachDone(e) {
+      this._doBackendSearch(e.target.value.toLowerCase());
+    }
+    _doBackendSearch(search) {
+      this._fire('replan', {search: (search == '*' ? null : search)});  // trigger planner
+
+      this._updateSearchProps(search);
+    }
+    _updateSearchProps(search) {
+      this._setProps(Object.assign(this._props || {}, {search, open: true}));
```

Comment:
An object should never call `_setProps` on itself. `_props` are only ever input to this object from outside. Instead, use `_setState` to track local data. Where `_props` are only ever affected by callers, `_state` should only ever be used by `this`.

Reply:
done